src/jdk.compiler/share/classes/com/sun/tools/javac/parser/JavacParser.java

```diff
@@ -3379,7 +3379,7 @@ JCExpression annotationValue() {
                     buf.append(annotationValue());
                 }
             }
-            accept(RBRACE);
+            accept(RBRACE, tk -> Errors.AnnotationMissingElementValue);
```

Comment:
I think that this method needs some more massage, annotations like this one are being accepted:

```
@SuppressWarnings({,})
```
even previous to your patch, if my reading of the spec is correct, there should be at least one `ElementValue`

Reply:
Indeed, that was my observation as well but then I noticed this: 

https://bugs.openjdk.java.net/browse/JDK-8012722

This is among the positive tests of that contribution:

```
public class SingleCommaAnnotationValue {
    @Foo({}) void a() { }
    @Foo({,}) void b() { }
    @Foo({0}) void c() { }
    @Foo({0,}) void d() { }
    @Foo({0,0}) void e() { }
    @Foo({0,0,}) void f() { }
}
@interface Foo { int[] value(); }
```

Reply:
To my understanding the following rule permits `{,}`:

```
ElementValueArrayInitializer:
   { [ElementValueList] [,] }
ElementValueList:
   ElementValue {, ElementValue}
```

Reply:
I see, thanks for the clarification